src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,222 @@
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled), typeof(WindowsOnly))]
+        [InlineData("winforms")]
+        [InlineData("winformslib")]
+        [InlineData("winformscontrollib")]
+        [InlineData("wpf")]
+        [InlineData("wpflib")]
+        [InlineData("wpfcustomcontrollib")]
+        [InlineData("wpfusercontrollib")]
+        public async Task VisualBasicTemplateProject_WindowsOnly_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [InlineData("classlib")]
+        [InlineData("console")]
+        [InlineData("mstest")]
+        [InlineData("nunit")]
+        [InlineData("xunit")]
+        public async Task VisualBasicTemplateProject_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = (templateName, isWindows: ExecutionConditionUtil.IsWindows) switch
+            {
+                (_, isWindows: false) => new string[] { ERR_UndefinedType1 },
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic, ignoredDiagnostics);
+        }
+
+        private async Task AssertTemplateProjectLoadsCleanlyAsync(string templateName, string languageName, string[] ignoredDiagnostics)
+        {
+            try
+            {
+                if (ignoredDiagnostics.Length > 0)
+                {
+                    TestOutputHelper.WriteLine($"Ignoring compiler diagnostics: \"{string.Join("\", \"", ignoredDiagnostics)}\"");
+                }
+
+                // Clean up previous run
+                CleanupProject(templateName, languageName);
+
+                var projectFilePath = GenerateProjectFromTemplate(templateName, languageName, TestOutputHelper);
+
+                await AssertProjectLoadsCleanlyAsync(projectFilePath, ignoredDiagnostics);
+
+                // Clean up successful run
+                CleanupProject(templateName, languageName);
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        private string GenerateProjectFromTemplate(string templateName, string languageName, ITestOutputHelper outputHelper)
+        {
+            var projectPath = GetProjectPath(templateName, languageName);
+            var projectFilePath = GetProjectFilePath(projectPath, languageName);
+
+            var exitCode = NewProject(templateName, projectPath, languageName, outputHelper);
+            Assert.Equal(0, exitCode);
```

Comment:
Consider moving the Assert.Equal() into NewProject so nobody forgets to check it.

Reply:
Moved